cloudify_types/cloudify_types/component/operations.py

```diff
@@ -66,16 +66,52 @@ def _get_desired_operation_input(key, args):
 def _get_client(kwargs):
     client_config = _get_desired_operation_input('client', kwargs)
     if client_config:
-        return CloudifyClient(**client_config)
+        client = CloudifyClient(**client_config)
+
+        # for determining an external client:
+        manager_ips = [mgr.private_ip for mgr in
```

Comment:
I think let's actually do both private_ip + public_ip into the set. It's our local manager in either case!